compiler/src/iree/compiler/Dialect/Flow/Transforms/RaiseSpecialOps.cpp

```diff
@@ -27,6 +28,103 @@ namespace Flow {
 
 namespace {
 
+LogicalResult tryRaiseSlicingOp(linalg::LinalgOp linalgOp,
+                                RewriterBase &rewriter) {
+  if (!linalgOp.hasTensorSemantics())
+    return failure();
+  // Check if there is only one result.
+  if (!llvm::hasSingleElement(linalgOp.getOperation()->getResults()))
+    return failure();
+  // Check if operation is elementwise. For now, we do not handle elementwise
+  // with permutations.
+  if (!isElementwise(linalgOp))
+    return failure();
+  if (!linalgOp.getIndexingMapsArray().back().isIdentity())
+    return failure();
+
+  // The outs should be a tensor.
+  auto initTensor =
+      dyn_cast<TensorType>(linalgOp.getDpsInitOperand(0)->get().getType());
+  if (!initTensor)
+    return failure();
+
+  linalg::YieldOp yieldOp =
+      cast<linalg::YieldOp>(linalgOp.getBlock()->getTerminator());
+
+  if (!yieldOp.getOperand(0).getDefiningOp())
+    return failure();
+
+  auto extractOp =
+      dyn_cast<tensor::ExtractOp>(yieldOp.getOperand(0).getDefiningOp());
+  if (!extractOp)
+    return failure();
+
+  // Get the tensor being extracted from.
+  TypedValue<RankedTensorType> source = extractOp.getTensor();
+
+  // Slicing does not change rank.
+  if (initTensor.getRank() == source.getType().getRank())
+    return failure();
+
+  // Check that the extract op is extracting a tensor defined outside
+  // the linalg op.
+  if (source.getDefiningOp() &&
+      source.getDefiningOp()->getParentOp() == linalgOp)
+    return failure();
+
+  auto one = rewriter.getIntegerAttr(rewriter.getIntegerType(64), 1);
+  auto zero = rewriter.getIntegerAttr(rewriter.getIntegerType(64), 0);
+
+  int64_t rank = source.getType().getRank();
+  SmallVector<OpFoldResult> offsets;
+  SmallVector<OpFoldResult> size;
+  // We can assume strides of 1, since we checked for elementwise before.
+  SmallVector<OpFoldResult> strides(rank, one);
+
+  auto outShape = initTensor.getShape();
+  for (auto it : llvm::enumerate(extractOp.getIndices())) {
```

Comment:
nit: prefer using `structured binding declaration`.

```suggestion
  for (auto [idx, indexValue] : llvm::enumerate(extractOp.getIndices())) {
```